nsxt/resource_nsxt_cluster_virtual_ip.go

```diff
@@ -0,0 +1,114 @@
+/* Copyright © 2023 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt-mp/nsx/cluster"
+	nsxModel "github.com/vmware/vsphere-automation-sdk-go/services/nsxt-mp/nsx/model"
+)
+
+var ClusterVirtualIPForceType = []string{
+	nsxModel.ClusterVirtualIpProperties_FORCE_TRUE,
+	nsxModel.ClusterVirtualIpProperties_FORCE_FALSE,
+}
+
+var DefaultIPv4VirtualAddress = "0.0.0.0"
+
+var DefaultIPv6VirtualAddress = "::"
+
+func resourceNsxtClusterVirualIP() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtClusterVirualIPCreate,
+		Read:   resourceNsxtClusterVirualIPRead,
+		Update: resourceNsxtClusterVirualIPUpdate,
+		Delete: resourceNsxtClusterVirualIPDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"force": {
+				Type:         schema.TypeString,
+				Description:  "On enable it ignores duplicate address detection and DNS lookup validation check",
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice(ClusterVirtualIPForceType, false),
+				Default:      nsxModel.ClusterVirtualIpProperties_FORCE_FALSE,
+			},
+			"ip_address": {
+				Type:         schema.TypeString,
+				Description:  "Virtual IPv4 address",
+				Optional:     true,
+				ValidateFunc: validation.IsIPv4Address,
+				Default:      DefaultIPv4VirtualAddress,
```

Comment:
Are those defaults equivalent to "unspecified"?

Reply:
Yes, so if we don't configure virtual ip of a cluster, its virtual ip property will be ipv4: 0.0.0.0, ipv6: ::